velox/vector/BaseVector.cpp

```diff
@@ -725,6 +726,7 @@ void BaseVector::prepareForReuse() {
       rawNulls_ = nullptr;
     }
   }
+  this->resetDataDependentFlags(SelectivityVector(this->size()));
```

Comment:
> SelectivityVector(this->size())

This memory allocation is not free. asciiSetRows_.deselect(rows); is not free either. Can we avoid these? Perhaps, change the API to pass raw pointer to SelectivityVector and allow it to be nullptr?